### DISCUSSION TITLE
[Feature Request] Simfiles library export

### DISCUSSION THREAD

**Discussion by SHRMP0** [TeamRizu/OutFox#252] [03 - Feature Requests]
_(this is a repost from the [forum](https://discourse.projectmoon.dance/t/simfiles-library-export/73), when this public issue tracker didn't exist yet)_

One of the biggest hassles for anyone with a big simfile library on events (or even on livestreams if your viewers can request songs) is letting people figure out if something is available on your machine, what charts are available and where to find them.

So, I’d like to request the option to export the simfile library to a document (for example: .csv, .xlsx, .docx, .pdf, .txt…), alongside some sorting and filtering options and the ability to include the group/song banners in order to give more creative freedom on how to publish it online or even print a physical copy.

I do understand that external tools/scripts capable of doing this are available, but it would be really nice to see this feature implemented directly inside the game, assuming that the engine can handle this kind of task.

**Comment by deltaryz** [comment]
Personally I would really appreciate the ability to add songs to "playlists", with full capability to edit and sort those playlists within the game's UI as well as externally via a config file.

The closest thing I could find with a quick Google search is this:
https://www.reddit.com/r/Stepmania/comments/7ipb70/how_do_i_make_a_custom_difficulty_nonrandom/

It seems that there are already config files for these things locally, shouldn't be too difficult to retrofit those into something more portable.
